src/test/java/org/dataloader/DataLoaderTest.java

```diff
@@ -737,22 +737,22 @@ public void should_Clear_objects_with_complex_key() throws ExecutionException, I
     public void should_Accept_objects_with_different_order_of_keys() throws ExecutionException, InterruptedException {
         List<Collection<JsonObject>> loadCalls = new ArrayList<>();
         DataLoaderOptions options = newOptions().setCacheKeyFunction(getJsonObjectCacheMapFn());
-        DataLoader<JsonObject, Integer> identityLoader = idLoader(options, loadCalls);
+        DataLoader<JsonObject, JsonObject> identityLoader = idLoader(options, loadCalls);
 
         JsonObject key1 = new JsonObject().put("a", 123).put("b", 321);
         JsonObject key2 = new JsonObject().put("b", 321).put("a", 123);
 
         // Fetches as expected
 
-        CompletableFuture<Integer> future1 = identityLoader.load(key1);
-        CompletableFuture<Integer> future2 = identityLoader.load(key2);
+        CompletableFuture<JsonObject> future1 = identityLoader.load(key1);
+        CompletableFuture<JsonObject> future2 = identityLoader.load(key2);
         identityLoader.dispatch();
 
         await().until(() -> future1.isDone() && future2.isDone());
         assertThat(loadCalls, equalTo(singletonList(singletonList(key1))));
         assertThat(loadCalls.size(), equalTo(1));
         assertThat(future1.get(), equalTo(key1));
-        assertThat(future2.get(), equalTo(key1));
+        assertThat(future2.get(), equalTo(key2));
```

Comment:
Sorry - why is this test changed for Java 11 ?  was it wrong?  and if so - why was it passing ?

Reply:
A brief inspection indicates the test shouldn't have been working with Java 8. We can see this because the `identityLoader` doesn't have identical types for its `K` and `V` type parameters - if we pass in a `JsonObject` to a loader generated from `idLoader`, we should expect to receive that same `JsonObject` coming out, but that's not what this test was checking.

As to why it was passing at all - I suspect it has something to do with this [bug](https://bugs.openjdk.org/browse/JDK-8058199) that was fixed with Java 9. Without digging into the `javac`-generated bytecode, it looks like a class check was missing from bytecode generated with Java 8. As such, when we tried to run with this code with Java 11, we started failing because we tried to cast a `JsonObject` (the returned value) to an `Integer` (the `V` in the `DataLoader`).

As to why I changed it from `key1` to `key2` - while `key1.equals(key2)` is true, it seemed neater to check we got `key2` for the future (as it corresponded to `future2`).

Reply:
Yeah I just looked into it - <K,V> are all just Object at runtime and since we never treated a value as a JsonObject - well the test got away with it.  Makes sense to fix